### PR TITLE
remove useless log entries concerning wrong version

### PR DESCRIPTION
**Pull request by hannesa2**
It removes this useless log entries, it concern to beta-version. This entries were miss leading.
Now it writes out a **warning** only on wrong real versionCode

image

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java [hunk 1]
```diff
@@ -160,16 +161,24 @@ public Object createFromParcel(Parcel source) {
...
+        int versionCode = versionBetaCodeFromString(version);
+        if (versionCode == 0) {
+            Log_OC.w(TAG, "Version string is incorrect " + version);
+            return 0;
+        }
+
+        return versionCode;
+    }
+
+    private static int versionBetaCodeFromString(String version) {
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java]: Why do you need this new method to remove the useless log entries? I would keep `versionCodeFromString` as it was before but adding just a new check `(versionCode == 0)` to log or not. Besides, you are not checking `versionCode == 0` for those direct calls to `versionBetaCodeFromString`.

Comment (hannesa2) [reply owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java]: nop, with your suggestion I would ignore versionNumber=0, what's not correct
Current solution is a one check-code, but only warning for non-beta

File: owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java [hunk 2]
```diff
@@ -90,22 +89,23 @@
...
-        public FeatureItem(int image, int titleText, int contentText, String version, String betaVersion,
-                           boolean showOnInitialRun) {
+        FeatureItem(int image, int titleText, int contentText, String version, String betaVersion,
+                    boolean showOnInitialRun) {
             this.image = image;
             this.titleText = titleText;
             this.contentText = contentText;
             this.versionNumber = versionCodeFromString(version);
-            this.betaVersion = versionCodeFromString(betaVersion);
+            this.betaVersion = versionBetaCodeFromString(betaVersion);
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java]: I think that with this call you are ignoring `versionCode == 0` check

File: owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java (hunk 1 above)
Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/features/FeatureList.java]: @hannesa2 see https://github.com/owncloud/android/pull/2443#discussion_r256507431

**Comment by hannesa2** [comment]
This is my solution, which should help this project, and I want to prevent to apply review changes, I do no support, till my tested fix stop working and causes new issues.

So you have 4 options:

- add your solution
- make an own solution
- merge it, like it is
- close it unmerged

I'm done and I'm fine with any solution

**Comment by davigonz** [comment]
Ok, let's give it a try, code approved, ready to test.

**Comment by jesmrec** [comment]
Two issues to comment here:

1.- By installing the app from scratch, the mentioned log `Version string is incorrect` is there, caused by the welcome wizard.
2- If you delete the history in the log view, no more logs are printed anymore.

**Comment by jesmrec** [comment]
sorry, behaviour 2.- also happens in `master`  branch. I will open a new issue.

**Comment by hannesa2** [comment]
image

I pushed branch to ˚upstream` remote. So you can do what you want

I'm done here

**Comment by jesmrec** [comment]
Ok, thanks a lot for your effort, but the PR did not pass the CR & QA needed for moving forward. Let's check another ones then.

**Pull request closed without merging**